docs.md

```diff
@@ -98,249 +110,443 @@ val s = RequiredParam("foo")(req)
 In the example above, the `MyRequest` type may be used in both `Endpoint` and `RequestReader` without any exceptions, 
 since there is an implicit view `myReqEv` defined. See [demo][1] for the complete example of custom request types.
 
-### Request Reader Monad
 
-Finch has built-in request reader that implements the [Reader Monad][2] functional design pattern:
+### Request Reader
+
+Finch has a built-in request reader that implements the [Reader Monad][2] functional design pattern:
 * `io.finch.request.RequestReader` reads `Future[A]`
 
-The simplified signature of the `RequestReader` abstraction is similar to `Service` but with monadic API methods `map`
-and `flatMap`:
+Since the request readers read futures they might be chained together with regular Finagle services in a single
+for-comprehension. Thus, reading the request params is an additional monad-transformation in the program's data flow.
+This is extremely useful when a service should fetch and validate the request params before doing the real job and not
+do the job at all if the params are not valid. Request reader might throw a future exception and none of further
+transformations will be performed. Reader Monad is sort of a famous abstraction that is heavily used in Finch.
+
+
+#### Overview
+
+A typical `RequestReader` might look like this:
+
+```scala
+import io.finch.request._
+
+case class User(name: String, age: Int, city: String)
+    
+val user: RequestReader[User] = (
+  RequiredParam("name") ~
+  RequiredParam("age").as[Int].shouldNot("be less than 18"){ _ < 18 } ~
```

Comment:
I'l switch to the predefined rule. But the DSL-style does not compile without adding more parentheses, so I think this syntax is clearer.